sentry_streams/sentry_streams/adapters/arroyo/consumer.py

```diff
@@ -59,9 +61,29 @@ def build_strategy(self, commit: Commit) -> ProcessingStrategy[Any]:
         follow.
         """
 
-        def add_route(message: Message[KafkaPayload]) -> RoutedValue:
-            value = message.payload.value
-            return RoutedValue(route=Route(source=self.source, waypoints=[]), payload=value)
+        def add_route(message: Message[KafkaPayload]) -> Union[FilteredPayload, RoutedValue]:
+            filtered = False
+            headers: Headers = message.payload.headers
+            if self.header_filter:
+                if self.header_filter not in headers:
+                    filtered = True
+
+            if filtered:
+                return FilteredPayload()
+            else:
+                value = message.payload.value
+                try:
+                    schema: Codec[Any] = get_codec(self.stream_name)
+                except Exception:
+                    raise ValueError(f"Kafka topic {self.stream_name} has no associated schema")
+
+                now = time.time()
+                return RoutedValue(
+                    route=Route(source=self.source, waypoints=[]),
+                    payload=StreamsMessage(
+                        payload=value, headers=headers, timestamp=now, schema=schema
```

Comment:
We should pick up the timestamp of the message coming from Kafka at the beginning. Not the consumer timestamp.

Reply:
yeah it was a typo, done